### DISCUSSION TITLE
Out-of-Box Typescript Support

### DISCUSSION THREAD

**Discussion by shehi** [TanStack/table#3959] [Ideas]
Hi,

Is out-of-box Typescript support is planned for v8? I don't mean those `@types/*` namespace of libraries, which most of the time are incomplete or outdated. 

It is extremely difficult to use this lib without one in the year 2022.

**Comment by talatkuyuk** [comment]
I use the react-table v.8 with typescript without problem, and I didn't install any type package like @`types/react-table`. It supports typescript as far as I use.

But sometimes I struggle to pass some table variables (table, table instance, column, cell props etc.) to other functions/components, because I think the types are mostly overwritten with the row type (like User, Product etc.)

**Reply by SamuelMabonga** [reply]
I am trying to use it as well but it keeps  asking me install types or add a declaration file. Any pointers on how to go about this?

**Comment by shehi** [comment]
v8 beta4 already had TS support. Use it.
